src/Spectre.Data/Datasets/RoiDictionary.cs

```diff
@@ -0,0 +1,117 @@
+﻿/*
+    * RoiDictionary.cs
+    * Class representing a set of tool to manage ROIs.
+    * Allows getting ROIs by name, adding and removing by name.
+
+    Copyright 2017 Roman LIsak
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Spectre.Data.RoiIo;
+
+namespace Spectre.Data.Datasets
+{
+    /// <summary>
+    /// Class representing a set of tools to manage ROIs.
+    /// </summary>
+    /// <seealso cref="Spectre.Data.Datasets.IRoiDictionary" />
+    public class RoiDictionary : IRoiDictionary
+    {
+        private List<Roi> _roiDataset = new List<Roi>();
+        private RoiReader _roireader;
+        private string _directoryPath;
+
+        /// <summary>
+        /// Sets the directory path.
+        /// </summary>
+        /// <param name="directoryPath">The directory path.</param>
+        public void SetDirectoryPath(string directoryPath)
+        {
+            _directoryPath = directoryPath;
+            _roireader = new RoiReader(_directoryPath);
+        }
+
+        /// <summary>
+        /// Loads all Rois from directory to the list.
+        /// All elements in the list will be overwritten.
+        /// </summary>
+        public void LoadAllRois()
+        {
+            _roiDataset = _roireader.GetAllRoisFromDirectory();
```

Comment:
It's good to cache these in memory, but why user can't also get them all at once (`return _roiDataset;`)? On the other hand, we should also check, whether `_roiDataset` hasn't been loaded yet - such situation is possible, and at this point we're reloading all of this from disk again... Even, if we consider that content of the disk could change, we should not reload everything! I'd even rather dropped caching of ROIs in RAM for the favor of having this lightweight.